### PR TITLE
[MI-1866] Refactor i18n

### PR DESCRIPTION
**Pull request by kcasas**
@zendesk/mintegrations 

### Description
* refactor i18n to use [zendesk/client-i18n-tools](https://github.com/zendesk/client-i18n-tools) instead of [yahoo/react-intl](https://github.com/yahoo/react-intl)
screen shot 2017-11-07 at 12 13 18 pm

### References
* JIRA: https://zendesk.atlassian.net/browse/MI-1866

### TODO
* **After merge samson needs to be updated to use `yarn translate` instead of `zat translate`.**

### Risks
* [medium] might not render marketplace description and installation instructions
* [medium] might not translate strings properly

### CODE REVIEW

File: dist/manifest.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
       "chat_sidebar": "assets/iframe.html"
     }
   },
-  "defaultLocale": "en-US",
+  "defaultLocale": "en-us",
```
Comment (miogalang) [inline dist/manifest.json]: why does this need to be lowercase? Is this intentional? how did this work before?

File: src/translations/locales.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
+[
```
Comment (miogalang) [inline src/translations/locales.json]: is this automatically generated? is this something we need to maintain? can't we grab this list from some API? or from the tools?

File: tests/Components/Customer/Note/__snapshots__/note.test.js.snap [hunk 3]
```diff
@@ -4,9 +4,7 @@ exports[`Note component on truncates correctly 1`] = `
 <div
   className="u-mb-sm u-zeta"
 >
-  <span>
-    Customer Notes
-  </span>
+  Customer Notes
```
Comment (miogalang) [inline tests/Components/Customer/Note/__snapshots__/note.test.js.snap]: Why are you taking these out of spans?

File: src/translations/locales.json (hunk 2 above)
Comment (kcasas) [reply src/translations/locales.json]: No, the tool wants us to generate this by hand based on the supported locales.

File: tests/Components/Customer/Note/__snapshots__/note.test.js.snap (hunk 3 above)
Comment (kcasas) [reply tests/Components/Customer/Note/__snapshots__/note.test.js.snap]: The spans are produced by `FormattedMessage` from `yahoo/react-intl`. The spans don't actually affect the UI

Comment (miogalang) [reply tests/Components/Customer/Note/__snapshots__/note.test.js.snap]: screenshots please to prove they still look the same @kcasas

Comment (kcasas) [reply tests/Components/Customer/Note/__snapshots__/note.test.js.snap]: Added screenshot above.

**Comment by miogalang** [comment]
Squash commits too please @kcasas

File: src/Helpers/object.js [hunk 4]
```diff
@@ -13,4 +13,24 @@ function flatten(object) {
   return flattened;
 }
 
-module.exports = flatten;
+function unflatten(data) {
+  const result = {}
```
Comment (samgavinio) [inline src/Helpers/object.js]: Seems the file is not included in the linter? Missing a semi-colon here.

**Pull request merged**